src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -3505,16 +3505,18 @@ private BoundExpression BindNullCoalescingOperator(BinaryExpressionSyntax node,
             return GenerateNullCoalescingBadBinaryOpsError(node, leftOperand, rightOperand, Conversion.NoConversion, diagnostics);
         }
 
-
         private BoundExpression BindNullCoalescingAssignmentOperator(AssignmentExpressionSyntax node, DiagnosticBag diagnostics)
         {
             BoundExpression leftOperand = BindValue(node.Left, diagnostics, BindValueKind.CompoundAssignment);
             BoundExpression rightOperand = BindValue(node.Right, diagnostics, BindValueKind.RValue);
 
+            // If either operand's Type is dynamic, then we potentially need to check overflow at runtime
+            bool isChecked = leftOperand.HasDynamicType() || rightOperand.HasDynamicType() ? CheckOverflowAtCompileTime : false;
```

Comment:
Consider parens for clarity of precedence.

Reply:
It may be simpler, and just as good, to store `CheckOverflowAtRuntime` into the bound node's `IsChecked`.

---
In reply to: [201185971](https://github.com/dotnet/roslyn/pull/28253#discussion_r201185971) [](ancestors = 201185971)

Reply:
Will address in a follow up. I wanted to replicate the behavior of `WithOverloadChecksIfApplicable`.

---
In reply to: [201188147](https://github.com/dotnet/roslyn/pull/28253#discussion_r201188147) [](ancestors = 201188147,201185971)